### PR TITLE
Fixed Images not announcing in VoiceOver

### PR DESCRIPTION
**Pull request by graytmatterMS**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Images were not being announced as Images by Voiceover. This is unexpected behavior and would have caused visually impaired users to miss images in our experience.

## Changelog

[MacOS] [Fixed] - Fixed Voiceover reading of Image 

## Test Plan

1. Go into RNTester/js/examples/Image/ImageExample.js and add accessible={true} accessibilityLabel="foo" to 'Plain Network Image' on line 415.
2. Compile and run app.
3. Navigate to Image page
4. Turn on Voiceover.
5. Navigate to the Plain Network Image.
- You should hear/see voiceover announce "foo, image"

 
Screen Shot 2020-06-26 at 14 34 36

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/475)

### CODE REVIEW

File: Libraries/Image/RCTImageView.m [hunk 1]
```diff
@@ -619,6 +619,10 @@ - (void)windowDidChangeBackingProperties:(NSNotification *)notification
   [self reloadImage];
 }
   
+- (id)reactAccessibilityElement
+{
+  return _imageView.cell;
+}
 - (NSColor *)tintColor
```
Comment (tom-un) [inline Libraries/Image/RCTImageView.m:626]: nit: newline

File: React/Views/UIView+React.m [hunk 2]
```diff
@@ -326,7 +326,7 @@ - (CGRect)reactContentFrame
 
 #pragma mark - Accessibility
 
-- (RCTPlatformView *)reactAccessibilityElement // TODO(macOS ISS#2323203)
```
Comment (tom-un) [inline React/Views/UIView+React.m:329]: You also need to update `UIView+React.h` and change the declaration there.

File: React/Views/UIView+React.h [hunk 3]
```diff
@@ -113,7 +113,7 @@
  * transparent in favour of some subview.
  * Defaults to `self`.
  */
-@property (nonatomic, readonly) RCTPlatformView *reactAccessibilityElement; // TODO(macOS ISS#2323203)
+@property (nonatomic, readonly) id *reactAccessibilityElement; // TODO(macOS ISS#2323203)
```
Comment (tom-un) [inline React/Views/UIView+React.h:116]: `id` is a pointer, so it should be `id reactAccessiblityElement` instead of `id *reactAccessibilityElement`

Comment (graytmatterMS) [reply React/Views/UIView+React.h]: My bad.

**Comment by tom-un** [comment]
As we discussed offline @graytmatterMS I can take over this PR (you can grant me permission to push directly to your branch, or I can make my own).

Changing the returning type to `id` breaks the iOS compilation :(.   I suggested offline that we change from `id` to `UIAccessibliy *` and `NSAccessibliy *` as a type for the property, but that has its own set of issues: both of those are "informal protocols" and there are other compilation breaks when properties are dereferenced.   So, I think the best approach is to just keep the return type, and cast the return in the new override in RCTImageView.m

**Comment by graytmatterMS** [comment]
> As we discussed offline @graytmatterMS I can take over this PR (you can grant me permission to push directly to your branch, or I can make my own).
> 
> Changing the returning type to `id` breaks the iOS compilation :(. I suggested offline that we change from `id` to `UIAccessibliy *` and `NSAccessibliy *` as a type for the property, but that has its own set of issues: both of those are "informal protocols" and there are other compilation breaks when properties are dereferenced. So, I think the best approach is to just keep the property type as-is, and cast the return in the new override in RCTImageView.m

I've added you as a collaborator to my repository. :)

**Pull request merged**